login/login.c

```diff
@@ -243,33 +243,45 @@ void login_syslog(glome_login_config_t* config, pam_handle_t* pamh,
   }
 }
 
-// read_stdin reads printable characters from stdin into buf. It returns:
+// read_stdin reads characters from stdin into buf. It returns:
 // -1, if it encounters an error while reading
-// -2, if it encounters EOF
 // (buflen-1) if it read buflen-1 characters
 // <(buflen-1), if a newline was read before the buffer was full
 // If the return value is >=0, the buf is NULL-terminated.
 static int read_stdin(char* buf, int buflen) {
-  int i = 0;
+  // Return error if we got no characters.
+  if (fgets(buf, buflen, stdin) == NULL) {
```

Comment:
We will need to adjust the calling code to allocate more than `ENCODED_BUFSIZE(GLOME_MAX_TAG_LENGTH)` to account for the 0 byte, but that bug has been here before.

Reply:
It seems not, `ENCODED_BUFSIZE` already contains `+1` for the 0 byte, so that should work fine.

Reply:
I keep being surprised by that API.